nipype/interfaces/bids.py

```diff
@@ -150,7 +154,18 @@ def _list_outputs(self):
 
         outputs = {}
         for key, query in self.inputs.output_query.items():
-            outputs[key] = layout.get(
-                **dict(query.items() | filters.items()),
-                return_type='file')
+            args = query.copy()
+            args.update(filters)
+            filelist = layout.get(return_type='file',
+                                      **args)
+            if len(filelist) == 0:
+                msg = 'Output key: %s returned no files' % (
+                    key)
+                if self.inputs.raise_on_empty:
+                    raise IOError(msg)
+                else:
+                    warn(msg)
```

Comment:
I would advocate here for using nipype's logging:

At the top:

```
from .. import logging
(...)

LOGGER = logging.getLogger('workflows')
```

And here:

```
LOGGER.warning('Output key: %s returned no files', key)
```